lib/CPANRatings/Model/SearchCPAN.pm

```diff
@@ -9,130 +9,133 @@ use URI::Escape ();
 my $json = JSON->new->utf8;
 
 sub new {
-  my ($proto, %args) = (shift, @_);
-  my $class = ref $proto || $proto;
-  my $self = { };
-  bless( $self, $class);
+    my ($proto, %args) = (shift, @_);
+    my $class = ref $proto || $proto;
+    my $self  = {};
+    bless($self, $class);
 }
 
 sub search_distribution {
-  shift->_search('dist', @_);
+    shift->_search('dist', @_);
 }
 
 sub _search {
-  my ($self, $mode, $query) = @_;
+    my ($self, $mode, $query) = @_;
 
-  return unless $query;
+    return unless $query;
 
-  my $cache = Combust::Cache->new(type => "CR.search");
+    my $cache = Combust::Cache->new(type => "CR.search");
 
-  my $data = $cache->fetch(id => "search.module;$mode;$query");
+    my $data = $cache->fetch(id => "search.module;$mode;$query");
 
-  if ($data) {
-    $data = $data->{data};
-  }
-  else {
-    my $url = "http://search.cpan.org/search?query=$query&mode=$mode&format=xml";
-    #warn "fetching $url from search.cpan";
-    $data = get $url;
-    my @results;
     if ($data) {
-      my @results; 
-
-      my $xp = XML::XPath->new(xml => $data);
-      foreach my $module ($xp->find("/results/$mode")->get_nodelist){
-	#print $module->find('name')->string_value, "\n";
-	#print $module->find('version')->string_value, "\n";
-	my $author_link = $module->find('author/link')->string_value;
-	my $author = { link => $author_link };
-	$author->{id} = $author_link;
-	$author->{id} =~ s!.*author/([A-Z]+)/$!$1!;
-	$author->{id} =~ s!.*/~([a-zA-Z]+)/$!$1!;
-
-        my $link = $module->find('link')->string_value;
-        $link = URI::Escape::uri_unescape($link);
-
-	my $distribution = $link;
-	$distribution =~ s!(.*?author/[A-Z]+/([^/]+)).*!$2!;
-	my $distribution_link = $1 ? $1 . "/" : '' ;
-	$distribution =~ s!-\d+(\.\d+)?(_\d+)?$!!;
-
-        my $name = $module->find('name')->string_value;
-
-	push @results, { name    => $name,
-			 description => $module->find('description')->string_value,
-			 version => $module->find('version')->string_value,
-			 link    => $link,
-			 author  => $author,
-			 distribution => { name => $name,
-					   link => $distribution_link,
-					 },
-		       };
-      }
-      $data = \@results;
-
-      #use Data::Dump qw(pp);
-      #warn pp($data);
+        $data = $data->{data};
     }
     else {
-      $data = [];
+        my $url = "http://search.cpan.org/search?query=$query&mode=$mode&format=xml";
```

Comment:
Does this API actually still work?

Reply:
I thought it did because the site worked, but it's failing in a way that I will just leave it out instead.